src/kubectlUtils.ts

```diff
@@ -1,344 +1,369 @@
+    if (failed(getResult)) {
+        throw new Error('Kubectl command failed: ' + getResult.error[0]);
+    }
+    return getResult.result;
+}
+
+/**
+ * Wait and block until the specified pod's status becomes running.
+ *
+ * @param kubectl the kubectl client.
+ * @param podName the pod name.
+ */
+export async function waitForRunningPod(kubectl: Kubectl, podName: string): Promise<void> {
+    while (true) {
+        const shellResult = await kubectl.invokeAsync(`get pod/${podName} --no-headers`);
+        if (shellResult.code !== 0) {
+            throw new Error(`Failed to get pod status: ${shellResult.stderr}`);
+        }
+        const status = shellResult.stdout.split(/\s+/)[2];
+        kubeChannel.showOutput(`pod/${podName} status: ${status}`);
+        if (status === "Running") {
+            return;
+        } else if (!isTransientPodState(status)) {
+            const logsResult = await kubectl.invokeAsync(`logs pod/${podName}`);
+            kubeChannel.showOutput(`Failed to start the pod "${podName}". Its status is "${status}".
+                Pod logs:\n${logsResult.code === 0 ? logsResult.stdout : logsResult.stderr}`);
+            throw new Error(`Failed to start the pod "${podName}". Its status is "${status}".`);
+        }
+
+        await sleep(1000);
+    }
+}
+
+function isTransientPodState(status: string): boolean {
+    return status === "ContainerCreating" || status === "Pending" || status === "Succeeded";
+}
+
+/**
+ * Get the specified resource information.
+ *
+ * @param kubectl the kubectl client.
+ * @param resourceId the resource id.
+ * @return the result as a json object, or undefined if errors happen.
+ */
+export async function getResourceAsJson<T extends KubernetesResource | KubernetesCollection<KubernetesResource>>(kubectl: Kubectl, resourceId: string, resourceNamespace?: string): Promise<T | undefined> {
+    const nsarg = resourceNamespace ? `--namespace ${resourceNamespace}` : '';
+    const shellResult = await kubectl.asJson<T>(`get ${resourceId} ${nsarg} -o json`);
+    if (failed(shellResult)) {
+        vscode.window.showErrorMessage(shellResult.error[0]);
+        return undefined;
+    }
+    return shellResult.result;
+}
+
+export async function createResourceFromUri(uri: vscode.Uri, kubectl: Kubectl) {
+    await changeResourceFromUri(uri, kubectl, 'create', 'creating', 'created');
+}
+
+export async function deleteResourceFromUri(uri: vscode.Uri, kubectl: Kubectl) {
+    const result = await vscode.window.showWarningMessage('Are you sure you?', 'Delete', 'Cancel');
+    if (result === 'Delete') {
+        await changeResourceFromUri(uri, kubectl, 'delete', 'deleting', 'deleted');
+    }
+}
+
+export async function applyResourceFromUri(uri: vscode.Uri, kubectl: Kubectl) {
+    await changeResourceFromUri(uri, kubectl, 'apply', 'applying', 'applied');
+}
+
+async function changeResourceFromUri(uri: vscode.Uri, kubectl: Kubectl, verb: string, errorVerb: string, successVerb: string) {
```

Comment:
We might want to call the last two parameters something like `verbParticiple` and `verbPast`.  For example, I could see a long-running operation using a progress box of the form `Creating foo... please wait` which would use the participle in a non-error context.  But this could be a future refactoring if and when someone used those forms differently.

Reply:
done.